src/main/java/org/takes/misc/VerboseList.java

```diff
@@ -0,0 +1,205 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package org.takes.misc;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+
+/**
+ * Verbose List that wraps OutOfBoundsException with custom message.
+ *
+ * @author I. Sokolov (happy.neko@gmail.com)
+ * @version $Id$
+ * @param <T> Type of item
+ */
+@SuppressWarnings({"PMD.TooManyMethods", "PMD.PreserveStackTrace"})
```

Comment:
@Happy-Neko Bad idea to suppress this warning. We can avoid it (and preserve original exception) in this way:

```java
try {
// code that throws IndexOutOfBoundsException
} catch (IndexOutOfBoundsException ex) {
    IndexOutOfBoundsException thrown = new IndexOutOfBoundsException(ex.getMessage());
    thrown.initCause(ex);
    throw thrown;
}
```